pkg/cache/clusterqueue_test.go

```diff
@@ -306,7 +381,32 @@ func TestFitInCohort(t *testing.T) {
 	for name, tc := range cases {
 		t.Run(name, func(t *testing.T) {
 			defer features.SetFeatureGateDuringTest(t, features.LendingLimit, tc.enableLendingLimit)()
-			got := tc.cq.FitInCohort(tc.request)
+			cache := New(utiltesting.NewFakeClient())
+
+			for _, flavorName := range tc.flavorNames {
+				cache.AddOrUpdateResourceFlavor(utiltesting.MakeResourceFlavor(flavorName).Obj())
+			}
+
+			for _, cq := range tc.clusterQueue {
+				_ = cache.AddClusterQueue(context.Background(), cq)
+			}
+
+			snapshot := cache.Snapshot()
+
+			i := 0
+			for fr, v := range tc.usage {
+				admission := utiltesting.MakeAdmission("CQ")
+				quantity := resources.ResourceQuantity(fr.Resource, v)
+				admission.Assignment(fr.Resource, fr.Flavor, quantity.String())
+
+				wl := utiltesting.MakeWorkload(fmt.Sprintf("workload-%d", i), "default-namespace").ReserveQuota(admission.Obj()).Obj()
+
+				cache.AddOrUpdateWorkload(wl)
+				snapshot.AddWorkload(workload.NewInfo(wl))
```

Comment:
You may consider using this new method to add usage, and can skip creating a fake workload:
https://github.com/kubernetes-sigs/kueue/blob/86691f1265974b9b58634773c928804952787bdb/pkg/cache/clusterqueue_snapshot.go#L51-L53